cpp/tests/structure/mg_select_random_vertices_test.cpp

```diff
@@ -0,0 +1,240 @@
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());
+      handle_->get_comms().barrier();
+      hr_timer.start("MG Construct graph");
+    }
+
+    auto [mg_graph, mg_edge_weights, mg_renumber_map] =
+      cugraph::test::construct_graph<vertex_t, edge_t, weight_t, false, true>(
+        *handle_, input_usecase, false, true);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());
+      handle_->get_comms().barrier();
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    auto mg_graph_view = mg_graph.view();
+    auto mg_edge_weight_view =
+      mg_edge_weights ? std::make_optional((*mg_edge_weights).view()) : std::nullopt;
+
+    raft::random::RngState rng_state(comm_rank);
+
+    //
+    // Test sampling from a distributed set
+    //
+
+    {
+      // Generate distributed vertex set to sample from
+      srand((unsigned)time(NULL));
+      std::vector<vertex_t> h_given_set(rand() % mg_graph_view.local_vertex_partition_range_size() +
+                                        1);
+
+      for (int i = 0; i < h_given_set.size(); i++) {
+        h_given_set[i] = mg_graph_view.local_vertex_partition_range_first() +
+                         rand() % mg_graph_view.local_vertex_partition_range_size();
+      }
+
+      std::sort(h_given_set.begin(), h_given_set.end());
+      auto last = std::unique(h_given_set.begin(), h_given_set.end());
+      h_given_set.erase(last, h_given_set.end());
+
+      // Compute size of the distributed vertex set
+      int num_of_given_set = static_cast<int>(h_given_set.size());
+
+      rmm::device_uvector<int> d_num_of_given_set(1, handle_->get_stream());
+      raft::update_device(d_num_of_given_set.data(), &num_of_given_set, 1, handle_->get_stream());
+      handle_->get_comms().allreduce(d_num_of_given_set.data(),
+                                     d_num_of_given_set.data(),
+                                     1,
+                                     raft::comms::op_t::SUM,
+                                     handle_->get_stream());
+      raft::update_host(&num_of_given_set, d_num_of_given_set.data(), 1, handle_->get_stream());
+      auto status = handle_->get_comms().sync_stream(handle_->get_stream());
+      CUGRAPH_EXPECTS(status == raft::comms::status_t::SUCCESS, "sync_stream() failure.");
+
+      // Move the distributed vertex set to GPUs
+      std::optional<rmm::device_uvector<vertex_t>> d_given_set{std::nullopt};
+      d_given_set = rmm::device_uvector<vertex_t>(h_given_set.size(), handle_->get_stream());
+      raft::update_device(
+        (*d_given_set).data(), h_given_set.data(), h_given_set.size(), handle_->get_stream());
+
+      // Sampling size should not exceed the size of distributed vertex set
+      size_t select_count = num_of_given_set > select_random_vertices_usecase.select_count
+                              ? select_random_vertices_usecase.select_count
+                              : num_of_given_set / 2 + 1;
```

Comment:
`num_of_given_set / 2 + 1;`
This number looks a bit arbitrary. Any rationale behind this?